sys/log/full/src/log_fcb.c

```diff
@@ -86,9 +86,10 @@ log_fcb_start_append(struct log *log, int len, struct fcb_entry *loc)
          * oldest flash area which was erased. If yes, then move watermark to
          * beginning of current oldest area.
          */
-        if ((fcb_log->fl_watermark_off >= old_fa->fa_off) &&
-            (fcb_log->fl_watermark_off < old_fa->fa_off + old_fa->fa_size)) {
-            fcb_log->fl_watermark_off = fcb->f_oldest->fa_off;
+        if (fcb_offset_in_sector(fcb, fcb_log->fl_watermark_off, old_sec)) {
+            struct fcb_sector_info si;
```

Comment:
Sorry, another annoying coding style nitpick: this variable definition needs to go at the top of the function.